main/signer.go

```diff
@@ -56,35 +55,31 @@ type Signer struct {
 }
 
 // handle incoming messages, create, sign and send a ubirch protocol packet (UPP) to the ubirch backend
-func signer(ctx context.Context, s *Signer) error {
-	for {
-		select {
-		case msg := <-s.MessageHandler:
-			// buffer last previous signature to be able to reset it in case sending UPP to backend fails
-			prevSign, found := s.protocol.Signatures[msg.ID]
-			if !found {
-				prevSign = make([]byte, 64)
-			}
+func (s *Signer) chainer() error {
+	for msg := range s.MessageHandler {
```

Comment:
I like the range, simplifies the handling

Reply:
:)